lib/datasource/local/note_local_datasource.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:notes_app/database.dart';
+import 'package:notes_app/models/note_model.dart';
+
+final noteLocalProvider = Provider<NoteLocalDatasource>(
+  (ref) => NoteLocalDatasource(ref.read(databaseProvider))
+);
+
+class NoteLocalDatasource {
+  NoteLocalDatasource._();
+  static final NoteLocalDatasource _instance = NoteLocalDatasource._();
```

Comment:
Manually creating singleton here is redundant because the provider already taken that responsibility for us